proofs.go

```diff
@@ -49,7 +53,12 @@ func (proof *SparseMerkleProof) sanityCheck(spec *TreeSpec) bool {
 		return true
 	}
 
-	siblingHash := hashSerialization(spec, proof.SiblingData)
+	var siblingHash []byte
+	if spec.sumTree {
+		siblingHash = hashSumSerialization(spec, proof.SiblingData)
+	} else {
+		siblingHash = hashSerialization(spec, proof.SiblingData)
+	}
```

Comment:
@h5law Whenever I've seen this sort of pattern in the past, I've found that adding a small helper can help with readability and avoid a ton of unnecessary if/else in the code. If we ever have more than two configs (unlikely but just sharing for other things you'll work on), it makes it even easier to support 2+ configs.

```go
# Add this
func hashSiblingData(isSumTree bool, spec, siblingData) []byte {
	if isSumTree {
		return hashSumSerialization(spec, proof.SiblingData)
	} 
	return hashSerialization(spec, proof.SiblingData)
}

# The code becomes this
siblingHash = hashSiblingData(spec.sumTree, spec, siblingData)
```	

You can do this everywhere so I will leave one comment and ping you to see if it's something you want to do before reviewing further.

Reply:
Addressed